<p>
  Remove literal boolean values from conditional expressions to improve readability. Anything that can be tested for equality with a boolean value must itself be a boolean value, and boolean values can be tested atomically.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
  if (booleanVariable == true) { /* ... */ }
  if (booleanVariable != true) { /* ... */ }
</pre>

<h2>Compliant Solution</h2>
<pre>
  if (booleanVariable) { /* ... */ }
  if (!booleanVariable) { /* ... */ }
</pre>
