<p>Shared coding conventions allow teams to collaborate efficiently. This rule checks that all constant names match a provided regular expression.</p>

<h2>Noncompliant Code Example</h2>

<p>With the default regular expression <code>^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$</code>:</p>

<pre>
public static const first:String = "first";
</pre>

<h2>Compliant Solution</h2>

<pre>
public static const FIRST:String = "first";
</pre>
