<p>
  Shared some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that field names match a provided regular expression.
</p>

<h2>Noncompliant Code Example</h2>

<p>With the default regular expression <code>^[_a-z][a-zA-Z0-9]*$</code>:</p>

<pre>
class MyClass {
  public var my_field:int;
}
</pre>

<h2>Compliant Solution</h2>

<pre>
public class MyClass {
   public var myField:int;
}
</pre>
