<p>
  Unused parameters are misleading. Whatever the value passed to such parameters is, the behavior will be the same.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
function doSomething(a:int, b:int):void      // "b" is unused
{
  compute(a);
}
</pre>

<h2>Compliant Solution</h2>
<pre>
function doSomething(a:int):void      // "b" is unused
{
  compute(a);
}
</pre>

<h2>Exceptions</h2>
<p>The following cases are ignored</p>
<ul>
  <li>overriding methods.</li>
  <li>all methods in classes implementing one or more interfaces.</li>
  <li>methods which are empty or where the body consists of a single comment or a single throw statement (i.e. where the intention is apparently to simulate an abstract class).</li>
</ul>
<pre>
override function doSomething(a:int):void {    // ignored
  compute(a);
}

...

class AbstractSomething {
  public function doSomething(a:int) {  // ignored
    throw new IllegalOperationError("doSomething() is abstract");
  }

...

interface I {
  function action(a:int, b:int);
}

class C extends I {
  function action(a:int, b:int) { // ignored
    return doSomethignWith(a);
  }
}
</pre>

<h2>See</h2>

<ul>
  <li> MISRA C++:2008, 0-1-11
  </li><li> MISRA C:2012, 2.7
</li></ul>