<p>
  There are several reasons for a method not to have a method body:
</p>

<ul>
    <li>It is an unintentional omission, and should be fixed to prevent an unexpected behavior in production.</li>
    <li>It is not yet, or never will be, supported. In this case an <code>NotSupportedException</code> should be thrown.</li>
    <li>The method is an intentionally-blank override. In this case a nested comment should explain the reason for the blank override.</li>
</ul>

<h2>Noncompliant Code Example</h2>

<pre>
public override function doSomething():void {
}

public function doSomethingElse():void {
}
</pre>
<h2>Compliant Solution</h2>

<pre>
public override function doSomething():void {
  throw new IllegalOperationError("doSomething cannot be performed because ...");
}

public function doSomethingElse():void {
  //This method is empty because ...
}
</pre>
