<p>
  Any statement or block of statements can be identified by a label, but those labels should be used only on 'while', 'do-while' and 'for' statements. Using labels in any other context leads to unstructured, confusing code.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
myLabel:if (i % 2 == 0) {            // Noncompliant
  if (i == 12) {
    print("12");
    break myLabel;
  }
  print("Odd number, but not 12");
}
</pre>

<h2>Compliant Solution</h2>
<pre>
myLabel:for (i = 0; i < 10; i++) {   // Compliant
  print("Loop");
  break myLabel;
}
</pre>
