<p>
  Having multiple cases in a <code>switch</code> with the same condition is confusing at best. At worst, it's a bug that is likely to induce further bugs as the code is maintained.
</p>

<h2>Extended desc for languages with <code>break</code></h2>

<p>
  If the first case ends with a break, the second case will never be executed, rendering it dead code. Worse there is the risk in this situation that future maintenance will be done on the dead case, rather than on the one that's actually used.
</p>
<p>
  On the other hand, if the first case does not end with a break, both cases will be executed, but future maintainers may not notice that.
</p>

<h2>Noncompliant Code Example</h2>
<pre>
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
  case 1:  // Noncompliant
    //...
    break;
}
</pre>

<h2>Compliant Solution</h2>
<pre>
switch(i) {
  case 1:
    //...
    break;
  case 5:
    //...
    break;
  case 3:
    //...
    break;
}
</pre>

